packages/web/src/repositories/leaderboard/leaderboard-repository-impl.ts

```diff
@@ -0,0 +1,96 @@
+import { NetworkClient } from "@common/clients/network-client";
+import { CommonError } from "@common/errors";
+import { LeaderboardRepository } from "./leaderboard-repository";
+import { GetLeaderboardRequest, UpdateLeaderboardHiddenStateRequest } from "./request";
+import {
+  GetLeaderboardResponse,
+  GetLeaderboardByAddressResponse,
+  nullLeaderboardInfo,
+  UpdateLeaderboardHiddenStateResponse,
+} from "./response";
+
+export class LeaderboardRepositoryImpl implements LeaderboardRepository {
+  private networkClient: NetworkClient | null;
+
+  constructor(networkClient: NetworkClient | null) {
+    this.networkClient = networkClient;
+  }
+
+  public getLeaderboard = async (request: GetLeaderboardRequest): Promise<GetLeaderboardResponse> => {
+    if (!this.networkClient) {
+      throw new CommonError("FAILED_INITIALIZE_PROVIDER");
+    }
+
+    const queries = [
+      request.keyword !== undefined ? `keyword=${request.keyword}` : "",
+      request.limit !== undefined ? `limit=${request.limit}` : "",
+      request.page !== undefined ? `page=${request.page}` : "",
+    ];
+
+    const response = await this.networkClient.get<{ data: GetLeaderboardResponse }>({
+      url: `/leaderboard?${queries.filter(query => !!query).join("&")}`,
+    });
+
+    if (!response?.data?.data) {
+      return nullLeaderboardInfo;
+    }
+
+    const data: GetLeaderboardResponse = response.data.data;
+    return data;
```

Comment:
In this case, the type inference is automatic, so there's no problem with returning directly. If you apply this fix, you'll need to apply it to other functions as well.
```suggestion
    return response.data.data;
```

Reply:
Thank you for the refactoring suggestion. I'm going to review it and keep the current approach.

keep it this way for the following reasons:
1. Value of explicit type declaration: By explicitly declaring the GetLeaderboardResponse type, the structure and intent of the return data is immediately clear to anyone reading the code.

2. Codebase consistency: As you mentioned, this change would require modifying all of our other functions, and we feel that making extensive changes for a small optimisation is a low priority at this point.

3. Ease of debugging: Having an intermediate variable makes it easier to inspect the value when debugging, which helps with troubleshooting.